html/janus.js

```diff
@@ -2403,9 +2406,32 @@ function Janus(gatewayCallbacks) {
 						}
 					} else {
 						// Use getDisplayMedia
-						stream = await navigator.mediaDevices.getDisplayMedia(constraints);
+						// Figure out if it is a video or audio capture request
+						if(track.type === 'screen') {
+
+							//double check if audio capture is required (messy way)
+							for(const othertrack of tracks) {
+								if(othertrack.type === 'desktopAudio' && othertrack.capture) {
+									constraints.audio = othertrack.capture;
```

Comment:
For some reason this comment I added disappeared, so I'll add a new single comment again. I don't like this approach, as it breaks the way `tracks` is supposed to work. You're not meant to simply pick any other random audio track from that list and use it here: we explicitly conceived `tracks` to be a collection of independent tracks, that you can either ask separately, or in some cases group together, e.g., in a single `getUserMedia` (which makes sense for instance when you're just asking for mic and webcam). Asking things together is done via the concept of `group`, which is explained in the original PR #3003. What you'e doing breaks that assumption, and will cause undefined or unexpected behaviours when one uses `tracks` to capture many different tracks at the same time.

Reply:
Hello,

Thanks for the feedback on this, I tried to stick to the code style conventions better this time.
I've reworked the code to make 'screen' work along side 'video' and 'audio' in the groups:

- The 'screen' track type has its own default group name to avoid collisions with expected behaviors of default audio/video group
-  A choice is made to acquire stream from `getDisplayMedia` rather than `getUserMedia` if the track is of the 'screen' type or the track belongs to a group where 'screen' is the type of the other track.
- Current demo for screenshare works the same as before (reverted changes)
- Actually sharing screen with system audio now requires that the 'screen' track is in a group with an 'audio' track

I'm still evaluating combinations of tracks to ensure proper behavior, but I wanted to check if this would an acceptable direction for a solution.